src/libs/actions/Transaction.ts

```diff
@@ -572,6 +584,139 @@ function getAllTransactions() {
     return Object.keys(allTransactions ?? {}).length;
 }
 
+function changeTransactionsReport(transactionIDs: string[], reportID: string, transactionIDToReportActionAndThreadData: Record<string, string>) {
```

Comment:
We should be building the `transactionIDToReportActionAndThreadData` map in this method, so we can send it to the API with the IDs needed. The shape is as follows:

```
{
   [transactionID]: {transactionThreadReportID: string, transactionThreadCreatedReportActionID: string, movedReportActionID: string}
}
```
Note that we only include `transactionThreadReportID` and `transactionThreadCreatedReportActionID` if we are creating a new optimistic transactionThread, otherwise we can omit those

```suggestion
function changeTransactionsReport(transactionIDs: string[], reportID: string) {
```